### PR TITLE
chore: rename charms.yaml file to interface.yaml

### PR DESCRIPTION
**Pull request by IronCore864**
In ["overhaul to better support charmhub's interface pages"](https://github.com/canonical/charm-relation-interfaces/pull/148), all `charms.yaml` are renamed to `interface.yaml`. Update `pytest-interface-tester` to load `interface.yaml` instead of `charms.yaml`.

See more discussion [here](https://github.com/canonical/charm-relation-interfaces/pull/153#issuecomment-2179621096).

Manual tests are done by installing this package from local path:

```bash
$ python3 run_matrix.py --include saml
Failed to load module test_requirer: No module named 'test_requirer'
INFO:root:Running tests for interface: saml
INFO:root:Running tests for version: v0
INFO:root:Running tests for role: provider
INFO:root:Running 1 saml interface tests on: ['saml-integrator']...
INFO:root:Running tests for saml
INFO:root:Running tests for charm: saml-integrator
INFO:root:Preparing testing environment for: saml-integrator
INFO:root:Cloning: saml-integrator from (https://github.com/canonical/saml-integrator-operator@main)
INFO:root:Preparing venv for /tmp/charm-relation-interfaces-tests/saml-integrator
INFO:root:Installing dependencies in venv for /tmp/charm-relation-interfaces-tests/saml-integrator
INFO:root:Installed lxml==4.9.4
ops==2.14.0
pydantic==1.10.16
signxml==3.2.2

INFO:root:Generating test file for saml at /tmp/charm-relation-interfaces-tests/saml-integrator/tests/interface
INFO:root:Running tests for /tmp/charm-relation-interfaces-tests/saml-integrator
You seem to be using pydantic v1. Please upgrade to v2, as compatibility may be dropped in a future version of pytest-interface-tester.
=================================================================== test session starts ====================================================================
platform darwin -- Python 3.11.8, pytest-8.2.2, pluggy-1.5.0
rootdir: /tmp/charm-relation-interfaces-tests/saml-integrator
configfile: pyproject.toml
plugins: interface-tester-2.0.1
collected 1 item

../../../../tmp/charm-relation-interfaces-tests/saml-integrator/tests/interface/interface_test_saml.py .                                             [100%]

==================================================================== 1 passed in 1.64s =====================================================================
INFO:root:Result: PASSED
INFO:root:Running tests for role: requirer
INFO:root:No tests specified for saml/requirer; skipping...
+++ Results +++
{
  "saml": {
    "v0": {
      "provider": {
        "saml-integrator": true
      },
      "requirer": {}
    }
  }
}
```

Note: We probably need to release a new version for this module but I'm not sure about the version: It's a breaking change, so by SEMVER it shouldn't be v2.0.2, although it seems "right" since it's a minor change... I left the version string in `pyproject.toml` unchanged, still 2.0.1 now.

### CODE REVIEW

**Review by tonyandrewmeyer — APPROVED** [review]
Looks good to me, although I'm not super familiar with this.

In regards to the version question, if we're not going to support both names, then a major bump seems the correct thing to do. It allows pinning to work as expected.

Note that you do need to have the version change in some way, because there's a workflow that does a release on merge to main.

File: interface_tester/collector.py [hunk 1]
```diff
@@ -152,19 +153,19 @@ def get_schemas(file: Path) -> Dict[Literal["requirer", "provider"], Type[DataBa
 
 
 def _gather_charms_for_version(version_dir: Path) -> Optional[_CharmsDotYamlSpec]:
-    """Attempt to read the `charms.yaml` for this version sudir.
+    """Attempt to read the `interface.yaml` for this version sudir.
```
Comment (dimaqq) [inline interface_tester/collector.py:156]: Could "sudir" be a typo? Maybe it's "subdir" or "directory"? Probably not "su[do] dir"?

File: interface_tester/collector.py [hunk 2]
```diff
@@ -61,7 +62,7 @@ def __hash__(self):
 
 
 class _CharmsDotYamlSpec(TypedDict):
-    """Specification of the `charms.yaml` file each interface/version dir should contain."""
+    """Specification of the `interface.yaml` file each interface/version dir should contain."""
```
Comment (dimaqq) [inline interface_tester/collector.py:65]: Would it make sense to rename the class too?

Unless I've misunderstood something...

**Review by dimaqq — APPROVED** [review]

File: interface_tester/collector.py (hunk 1 above)
Comment (tonyandrewmeyer) [reply interface_tester/collector.py:156]: Or "version's dir"? Or `version_dir` to match the arg.

Comment (PietroPasotti) [reply interface_tester/collector.py:156]: also, please rename the `_CharmsDotYamlSpec` to ``_InterfacesDotYamlSpec` or something more telling

**Review by PietroPasotti — APPROVED** [review]
Looks good to me.  

As for the version bump, I think we're the only ones depending on this so probably we could get away with a minor bump without breaking anyone's code, but still, my personal philosophy is: versions are free. So let's give it a major bump for the sake of hygiene :)

**Comment by IronCore864** [comment]
Thanks all for the review, I have fixed the naming and bumped the version to 3.0.0 in `pyproject.toml`.

**Comment by IronCore864** [comment]
A gentle reminder: Could you please merge it? @PietroPasotti 

I don't have permission for this repo. Thanks!

**Comment by PietroPasotti** [comment]
Sorry! was on holiday some time :)

**Pull request merged**